platform/darwin/src/MGLOfflinePack.mm

```diff
@@ -135,9 +135,12 @@ - (void)suspend {
 - (void)invalidate {
     MGLLogInfo(@"Invalidating pack.");
     MGLAssert(_state != MGLOfflinePackStateInvalid, @"Cannot invalidate an already invalid offline pack.");
+    MGLAssert(self.mbglOfflineRegion, @"Should have a valid region");
 
     self.state = MGLOfflinePackStateInvalid;
-    _mbglFileSource->setOfflineRegionObserver(*self.mbglOfflineRegion, nullptr);
+    if (self.mbglOfflineRegion) {
```

Comment:
What do you think about using `synchronized`? it may be possible that a different thread nils `mbglOfflineRegion` when executing line 142

Reply:
Yep, good idea!